### PR TITLE
feat: enable `compress` by default

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [x] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
Yes

### Motivation / Use-Case
Enable `compress` by default

### Breaking Changes

Maybe

### Additional Info
No

### CODE REVIEW

**Comment by alexander-akait** [comment]
hm, do you think it should be `true` by default?

**Comment by snitin315** [comment]
Yes, I also see it in TODO https://github.com/webpack/webpack-dev-server/projects/1#card-19879364

**Review by alexander-akait — DISMISSED** [review]

**Review by anshumanv — COMMENTED** [review]
need update docs

**Review by alexander-akait — APPROVED** [review]

File: examples/cli/compression/README.md [hunk 1]
```diff
@@ -3,19 +3,38 @@
...
 2. Files being sent to the browser from the `webpack` bundle should be gzipped.
 3. Open the console in your browser's devtools and select the _Network_ tab.
 4. Find `bundle.js`. The response headers should contain `Content-Encoding: gzip`.
 
+## `--no-compress`
+
+To run this example, run this command in your console or terminal:
+
+```console
+npm run webpack-dev-server -- --open --no-compress
```
Comment (anshumanv) [inline examples/cli/compression/README.md:28]: we run webpack serve right?

Comment (snitin315) [reply examples/cli/compression/README.md:28]: Actually it is script https://github.com/webpack/webpack-dev-server/blob/88468b67ed3f66470acc7816a25e49e07a0fadbf/package.json#L32

**Review by anshumanv — APPROVED** [review]
nicenicenice

File: examples/cli/compression/README.md (hunk 1 above)
Comment (anshumanv) [reply examples/cli/compression/README.md:28]: looks good then

Comment (alexander-akait) [reply examples/cli/compression/README.md:28]: Let's change it to `npx webpack serve` in future PRs

Comment (anshumanv) [reply examples/cli/compression/README.md:28]: will take care

**Comment by Den-dp** [comment]
@snitin315 I thought that reading (and writing) the content served in the memory - is the fastest thing that can be possible.

And adding compression for the in-memory server will only lead to more CPU usage, and may slow down a feedback loop for the developer (longer refresh time).

In theory, after this change, we will pay more CPU load for in-memory content compression, and un-compression (by Express and the browser). 🤔

**Comment by alexander-akait** [comment]
@Den-dp yes, but in other case HMR and live reloading will be faster because you will get compressed sources, we collect feedback and maybe revisit it, but I think it is better to have fast development env

**Comment by Den-dp** [comment]
@alexander-akait looks like @sokra also hit a similar problem with enabling compression by default https://github.com/webpack/webpack/pull/13962

**Comment by alexander-akait** [comment]
it is another problem

**Pull request merged**